src/Core/Infrastructure/Configuration/MetaService/Repository/DbMetaServiceFactory.php

```diff
@@ -35,5 +35,62 @@ class DbMetaServiceFactory
      */
     public static function createFromRecord(array $data): MetaService
     {
+        return (new MetaService(
+            (int) $data['id'],
+            $data['name'],
+            self::normalizeCalculationType($data['calculation_type']),
+            (int) $data['meta_selection_mode'],
+            self::normalizeDataSourceType((int) $data['data_source_type'])
+        ))
+        ->setWarning(self::getIntOrNull($data['warning']))
+        ->setCritical(self::getIntOrNull($data['critical']))
+        ->setOutput($data['output'])
+        ->setMetric($data['metric'])
+        ->setActivated((int) $data['is_activated'] === 1)
+        ->setRegexpSearchServices($data['regexp_search_services']);
+    }
+
+    /**
+     * This function will normalize the calculation type coming from the database
+     *
+     * @param string|null $calculationType
+     * @return string
+     */
+    private static function normalizeCalculationType(?string $calculationType): string
+    {
+        switch ($calculationType) {
+            case 'AVE':
+                return MetaService::CALCULTATION_TYPE_AVERAGE;
+            case 'MIN':
+                return MetaService::CALCULTATION_TYPE_MINIMUM;
+            case 'MAX':
+                return MetaService::CALCULTATION_TYPE_MAXIMUM;
+            case 'SOM':
+                return MetaService::CALCULTATION_TYPE_SUM;
+            default:
+                return MetaService::CALCULTATION_TYPE_AVERAGE;
+        }
+    }
+
+    /**
+     * This function will normalize the data source type coming from the database
+     *
+     * @param int|null $dataSourceType
+     * @return string
+     */
+    private static function normalizeDataSourceType(?int $dataSourceType): string
+    {
+        switch ($dataSourceType) {
+            case 0:
+                return MetaService::DATA_SOURCE_GAUGE;
+            case 1:
+                return MetaService::DATA_SOURCE_COUNTER;
+            case 2:
+                return MetaService::DATA_SOURCE_DERIVE;
+            case 3:
+                return MetaService::DATA_SOURCE_ABSOLUTE;
+            default:
+                return MetaService::DATA_SOURCE_GAUGE;
+        }
```

Comment:
juste for info (PHP8) ;)
```suggestion
        return match ($dataSourceType) {
            0 => MetaService::DATA_SOURCE_GAUGE,
            1 => MetaService::DATA_SOURCE_COUNTER,
            2 => MetaService::DATA_SOURCE_DERIVE,
            3 => MetaService::DATA_SOURCE_ABSOLUTE,
            default => MetaService::DATA_SOURCE_GAUGE
        }
```